src/distribution_spatial.cpp

```diff
@@ -51,6 +51,54 @@ Position CartesianIndependent::sample() const
   return {x_->sample(), y_->sample(), z_->sample()};
 }
 
+//==============================================================================
+// SphericalIndependent implementation
+//==============================================================================
+
+SphericalIndependent::SphericalIndependent(pugi::xml_node node)
+{
+  // Read distribution for r coordinate
+  if (check_for_node(node, "r")) {
+    pugi::xml_node node_dist = node.child("r");
+    r_ = distribution_from_xml(node_dist);
+  } else {
+    // If no distribution was specified, default to a single point at r=0
+    double x[] {0.0};
+    double p[] {1.0};
+    r_ = UPtrDist{new Discrete{x, p, 1}};
```

Comment:
Let's use `make_unique` instead:
```suggestion
    r_ = std::make_unique<Discrete>(x, p, 1);
```